dcos/httpclient.go

```diff
@@ -0,0 +1,69 @@
+package dcos
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	userAgent = ClientName + " - " + Version
```

Comment:
this is imo to static we should have something that lets users of the lib set additional information 
like also the user agent stuff should support setting additonal stuff like `dcos/client-go 0.0.1 - dcos-cli 0.2.3` or `dcos/client-go 0.0.1 - dcos-terraformprovider 1.2.3`

Reply:
I would propose that setting the `User-Agent` is the responsibility of `*dcos.Client`. In general I think we should only add behaviour to the `Transport`/`RoundTripper` that other clients might want to use (i.e. if you pass in the `dcos.DefaultTransport` to your `NewMyCustomFoobarClient()`).

For me that's currently only authentication/authorization.

Things that the `*dcos.Client` handles should, IMHO, be: logging, convenience methods for URLs, content-type headers, etc.

But... it's a blurry line of course. In this case, though, I think `ClientName` already hints at the fact that it's the responsibility of the client? :)

Reply:
yep I agree on this. Was thinking about third party clients and their User-Agent. However I agree User-Agent should be handled on dcos.Client. Will delete it... was just a proposal